app/server.py

```diff
@@ -7,6 +9,47 @@ def refresh():
     writer.update_calendars()
     return '', 200
 
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    form = forms.LoginForm()
+    flask.flash(form.validate_on_submit())
+    if form.validate_on_submit():
+        user = models.User.query.filter_by(email=form.email.data).first()
+        if user is None:
+            return flask.redirect('/login')
+        login_user(user, remember=form.remember.data)
+
+        flask.flash('Logged in successfully.')
+
+        next_url = flask.request.args.get('next')
+        if not auth.next_is_valid(next_url):
+            return flask.abort(400)
+
+        return flask.redirect(next_url or '/')
+    return flask.render_template('login.html', form=form)
+
+@app.route('/logout', methods=['POST'])
+@login_required
+def logout():
+    logout_user()
+    return redirect('/login')
+
+@app.route('/register', methods=['GET', 'POST'])
+def register():
+    form = forms.RegistrationForm()
+    if form.submit.data and form.validate():
```

Comment:
We should either add a recaptcha or email validation to ensure that we don't get spammed.